### ISSUE TITLE
Too many functions are written in PL/pgsql

### ISSUE THREAD

**Issue by davidfetter** [leondutoit/pg-need-to-know#1]
A lot of them could be SQL, which can much more easily be inlined, and is easier to comprehend.

Would you be interested in a patch to do this refactoring?

**Comment by leondutoit** [comment]
Hi @davidfetter thanks for taking interest. Can you explain the inlining part? Is this a performance gain?

**Comment by davidfetter** [comment]
Partly performance, partly clarity of the code.

**Comment by leondutoit** [comment]
And apart from clarity of code, from your perspective, is there any significant performance gain in general? Any documentation about this? Do you plan on using this code?

**Comment by davidfetter** [comment]
In general, the gain I see is more in clarity than in performance, but there are some easy performance wins to be had, some by refactoring to be more declarative.

**Comment by leondutoit** [comment]
Right. This is a bit of a pet project, but I also intend to use it in production soon, so I'd like the API to be stable (although I was planning on returning json from most functions instead of text, since my primary use case is usage via a REST API).  But given that, it would be great if you were willing to submit a patch. I'm sure I'll learn something in the process.
